normandy/recipes/models.py

```diff
@@ -122,18 +163,52 @@ def update_signature(self):
 
     @transaction.atomic
     def update(self, force=False, **data):
```

Comment:
> This doesn't check all revisions, it only checks the latest_revision to make sure that it has changed.

Maybe I'm missing something? The code fetches revisions that match the latest revision's ID, and then filters out revisions that don't have the same m2m relationships, then filters out revisions that have the same data. Then it says that it's clean if there exist any revisions that match all of that.

None of those filters check if the revisions you're searching for are actually the last revision though. You could do all that filtering and match a revision from months ago that had the same content, and it'd be considered clean even if the latest revision is actually different. If you already have a reference to the latest revision, why do you need to go through all that querying?